src/renderers/testing/ReactShallowRenderer.js

```diff
@@ -100,26 +59,146 @@ class ReactShallowRenderer {
       element.type,
     );
 
-    // TODO Remove this wrapper if/when context parameter is removed
-    if (context && Object.keys(context).length) {
-      element = wrapElementWithContextProvider(element, context);
-    }
+    this._element = element;
 
-    this._renderer = ReactTestRenderer.create(element, {
-      createNodeMock: createShallowNodeMock(),
-    });
+    if (this._instance) {
+      this._rendered = updateClassComponent(
+        this._instance,
+        element.props,
+        context,
+      );
+    } else {
+      const prototype = element.type.prototype;
+
+      if (typeof prototype.render === 'function') {
```

Comment:
The first one tests for *factory components*. This is a bit different. See my example in the comment above. We support components that are functions but return instances.

Other examples depend on instances already existing. I was talking about how to determine something should be `new`'d *before it gets created*. It is important because it is unsafe to `new` some things (for example arrow functions), and because we support a pattern that doesn't need `new` but still returns instances (factory components).